common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -48,53 +48,63 @@ def versions_by_type(dependency)
 
       def ignore_patch(version)
         parts = version.split(".")
-        return [] unless parts.size > 2
-
-        lower_parts = parts.first(2) + ["a"]
-        upper_parts = parts.first(2)
-        upper_parts[1] = upper_parts[1].to_i + 1
+        version_parts = parts.fill(0, parts.length...4)
+        lower_parts = if numeric_version?(version)
+                        version_parts.first(3) + [version_parts[3].to_i + 1] + ["a"]
+                      else
+                        version_parts.first(2) + [version_parts[2].to_i + 1] + ["a"]
+                      end
+        upper_parts = if numeric_version?(version)
+                        version_parts.first(1) + [version_parts[1].to_i + 1]
+                      else
+                        version_parts.first(2) + [999_999]
+                      end
         lower_bound = ">= #{lower_parts.join('.')}"
         upper_bound = "< #{upper_parts.join('.')}"
         ["#{lower_bound}, #{upper_bound}"]
       end
 
       def ignore_minor(version)
         parts = version.split(".")
-        return [] if parts.size < 2
-
-        if Gem::Version.correct?(version)
-          lower_parts = parts.first(2) + ["a"]
-          upper_parts = parts.first(1)
-          lower_parts[1] = lower_parts[1].to_i + 1
-          upper_parts[0] = upper_parts[0].to_i + 1
-        else
-          lower_parts = parts.first(1) + ["a"]
-          upper_parts = parts.first(1)
-          begin
-            upper_parts[0] = Integer(upper_parts[0]) + 1
-          rescue ArgumentError
-            upper_parts.push(999_999)
-          end
-        end
-
+        version_parts = parts.fill(0, parts.length...3)
```

Comment:
Nit: this seems an awful lot like `ignore_patch(version)` now, but with `version_parts = parts.fill(0, parts.length...4)` is now `version_parts = parts.fill(0, parts.length...3)`.

Is it possible we could fold this back to a single function that accepts a `precision` parameter? https://github.com/dependabot/dependabot-core/pull/3513/commits/7375421618e2ffd4380f0c767f453a2d5bb5bb9f#diff-67a26681133898ab39f6dac25dad58b8bfa751fd7245ec723c196f1184f93101R54-R68